### DISCUSSION TITLE
Confirming PayPal subscription with pocketbase

### DISCUSSION THREAD

**Discussion by rymdmetafor** [pocketbase/pocketbase#2661] [Q&A]
Hi everyone!

I have a client side rendered react app and I am looking into implementing a payed membership with PayPal. The CMS (possibly pocketbase) will store the users personal files, which they can access and edit if they have an active membership.

I wonder if there is a way to change a record that stores the users subscription information (Active/Inactive) when the subscription status changes at PayPal. As I understand it, [PayPal has webhooks that can deliver this information](https://developer.paypal.com/api/rest/webhooks/event-names/#subscriptions).

Is it possible to read a webhook directly in pocketbase or will I have to write a function elsewhere to do this?

Thank you for your patience with such a simple question.

Sebastian

**Comment by ganigeorgiev** [comment]
You can [register a new Go route](https://pocketbase.io/docs/router/#register-new-route) to handle and verify the PayPal webhook. With the next release there will be an option to bind to the app hooks with JavaScript too.

Alternatively, you can handle the webhook in a 3rd party backend service (eg. node). The JS SDK can be used by both the regular app users and PocketBase admins (for an auth example you can check the [Authentication short guide](https://pocketbase.io/docs/authentication/)).

**Reply by tanzilai** [reply]
Gg
